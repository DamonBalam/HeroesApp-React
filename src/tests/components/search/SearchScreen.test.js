const { mount } = require('enzyme');
const { MemoryRouter, Route } = require('react-router-dom');
const { SearchScreen } = require('../../../components/search/SearchScreen');

describe('Pruebas en el SearchScreen', () => {
    

    test('Debe de mostrarse correctamente <SearchScreen />', () => {
        const wrapper = mount(
            <MemoryRouter initialEntries={['/search']}>
                <Route path='/search' component={SearchScreen} />
            </MemoryRouter>
        );
        expect(wrapper).toMatchSnapshot();

        expect( wrapper.find('.alert-info').text().trim() ).toBe('Search a Hero');
    });


    test('should mostrar a batman y en input del query', () => {
        const wrapper = mount(
            <MemoryRouter initialEntries={['/search?q=batman']}>
                <Route path='/search' component={SearchScreen} />
            </MemoryRouter>
        );
        
        expect( wrapper.find('input').prop('value') ).toBe('batman');
        expect( wrapper ).toMatchSnapshot();

    });
    
});
