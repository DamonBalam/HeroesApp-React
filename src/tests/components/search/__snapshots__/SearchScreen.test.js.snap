// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el SearchScreen Debe de mostrarse correctamente <SearchScreen /> 1`] = `
<div
  className="row"
>
  <div
    className="col-5"
  >
    <h4>
      Search Screen
    </h4>
    <hr />
    <form
      onSubmit={[Function]}
    >
      <input
        autoComplete="off"
        className="form-control"
        name="searchText"
        onChange={[Function]}
        placeholder="Find your hero"
        type="text"
        value=""
      />
      <button
        className="btn btn-block btn-outline-info mt-1"
        type="submit"
      >
        Search...
      </button>
    </form>
  </div>
  <div
    className="col-7"
  >
    <h4>
      Results
    </h4>
    <hr />
    <div
      className="alert alert-info animate__animated animate__fadeInRight"
    >
      Search a Hero
    </div>
  </div>
</div>
`;

exports[`Pruebas en el SearchScreen should mostrar a batman y en input del query 1`] = `
<div
  className="row"
>
  <div
    className="col-5"
  >
    <h4>
      Search Screen
    </h4>
    <hr />
    <form
      onSubmit={[Function]}
    >
      <input
        autoComplete="off"
        className="form-control"
        name="searchText"
        onChange={[Function]}
        placeholder="Find your hero"
        type="text"
        value="batman"
      />
      <button
        className="btn btn-block btn-outline-info mt-1"
        type="submit"
      >
        Search...
      </button>
    </form>
  </div>
  <div
    className="col-7"
  >
    <h4>
      Results
    </h4>
    <hr />
    <div
      className="card ms-3"
      style={
        Object {
          "maxWidth": 540,
        }
      }
    >
      <div
        className="row no-gutters"
      >
        <div
          className="col-md-4"
        >
          <img
            alt="imagen de super hero"
            className="card-img"
            src="./assets/heroes/dc-batman.jpg"
          />
        </div>
        <div
          className="col-md-8"
        >
          <div
            className="card-body"
          >
            <h5
              className="card-title"
            >
              Batman
            </h5>
            <p
              className="card-text"
            >
              Bruce Wayne
            </p>
            <p
              className="card-text"
            >
              <small
                className="text-muted"
              >
                Detective Comics #27
              </small>
            </p>
            <a
              className="btn btn-block btn-primary"
              href="/hero/dc-batman"
              onClick={[Function]}
            >
              Ver m√°s...
            </a>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
